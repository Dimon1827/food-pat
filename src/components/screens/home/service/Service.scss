@import '@/assets/styles/variables';

.service {
  padding-bottom: 40px;

  @media (min-width: $tablet-width) {
    color: $black-color;
    padding-bottom: 80px;
  }
}

.service__list {
  margin: 0;
  padding: 0;
  margin-top: 25px;
  list-style-type: none;

  @media (min-width: $tablet-width) {
    display: flex;
    gap: 30px;
  }

  @media (min-width: $desktop-width) {
    margin: 0;
    gap: 80px;
  }
}

.service__item {
  padding: 23px 20px 18px 20px;
  position: relative;
  border-radius: 20px;
  background-color: #fff3eb;

  &::after {
    content: "";
    z-index: 1;
    position: absolute;
    width: 161px;
    height: 112px;
    right: 0;
    top: 34%;
    background: url("./images/breakfast-mobile.png");
  }

  &:not(:last-child) {
    background-color: #f2fce4;
    margin-bottom: 20px;

    &::after {
      background: url("./images/vegetable-mobile.png");
      background-repeat: no-repeat;
    }
  }

  @media (min-width: $tablet-width) {
    padding: 46px 133px 39px 27px;
    width: 48%;
    min-height: 203px;

    &::after {
      right: 1px;
      top: 121px;
      width: 172px;
      height: 172px;
      background: url("./images/breakfast-tablet.png");
    }

    &:not(:last-child) {
      margin-bottom: 0;

      &::after {
      background: url("./images/vegetable-tablet.png");
      }
    }
  }

  @media (min-width: $desktop-width) {
    width: 570px;
    padding: 78px 133px 39px 76px;

    &::after {
      top: 154px;
      width: 240px;
      height: 170px;
    }

    &:not(:last-child) {
      margin-bottom: 0;
    }
  }
}

.service__figure {
  width: 67px;
  height: 50px;
  background-repeat: no-repeat;
  position: absolute;
  top: 0;
  left: -8px;

  @media (min-width: $tablet-width) and (max-width: $desktop-width) {
    width: 268px;
    height: 211px;
    position: absolute;
    left: 61%;
    top: 12%;
  }

  @media (min-width: $desktop-width) {
    width: 100px;
    height: 106px;
    left: 0;
    top: -34%;
  }
}

.service__description {
  margin: 0;
  color: #253d4e;
  width: 50%;
  font-size: 19px;
  line-height: 125%;
  margin-bottom: 19px;

  @media (min-width: $tablet-width) {
    width: 200px;

    font-size: 30px;
    padding: 0;
    margin-bottom: 14px;
    border: none;
  }

  @media (min-width: $desktop-width) {
    margin-bottom: 33px;
  }
}

.service__link {
  display: inline-block;
  position: relative;
  color: $white-color;
  background-color: #3bb77e;
  text-align: start;
  border-radius: 10px;
  width: 100px;
  padding: 12px;
  border: none;
  cursor: pointer;
  font-family: $oswald-font;
  font-size: 16px;
  line-height: 130%;
  letter-spacing: 0.04em;
  text-decoration: none;

  &::after {
    content: "";
    position: absolute;
    top: 18px;
    right: 7px;
    display: inline-block;
    width: 24px;
    height: 11px;
    background: url("./icons/arrow.svg");
    transition: all 0.2s;
  }

  &:active {
    opacity: 0.3;
  }

  &:hover,
  &:active {
    opacity: 0.3;

    &::after {
      right: 0;
      width: 31px;
      background-image: url("./icons/icon-width-arrow.svg");
    }
  }

  @media (min-width: $tablet-width) {
    font-size: 20px;
    width: 120px;
    line-height: 20px;

    &::after {
      top: 18px;
      right: 17px;
    }

    &:hover,
    &:active {
      &::after {
        right: 7px;
      }
    }
  }
}
