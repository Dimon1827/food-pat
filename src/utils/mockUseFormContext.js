import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


