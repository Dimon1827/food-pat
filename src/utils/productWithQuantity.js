const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
