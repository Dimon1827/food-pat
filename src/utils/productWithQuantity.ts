import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
