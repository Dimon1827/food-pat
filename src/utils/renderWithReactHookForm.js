import { render } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  let reactHookFormMethods = {};

  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}