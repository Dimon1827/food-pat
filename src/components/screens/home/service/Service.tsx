import { useNavigate } from 'react-router-dom';
import './Service.scss';

const serviceItems = [
  'Лучшие органические продукты онлайн',
  'Приготовьте свой завтрак Здоровый и легкий',
];
function Service() {
  const navigate = useNavigate();

  return (
    <section className="service">
      <h2 className="visually-hidden">Услуги</h2>
      <div className="container">
        <ul className="service__list">
          {serviceItems.map((item,index) => {
            return (
              <li className="service__item" key={index}>
                <p className="service__description">{item}</p>
                <button
                  onClick={() => navigate('/catalog')}
                  className="service__link"
                >
                  Купить
                </button>
              </li>
            );
          })}
        </ul>
      </div>
    </section>
  );
}

export default Service;
