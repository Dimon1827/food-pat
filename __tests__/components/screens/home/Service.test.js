import { fireEvent, screen } from '@testing-library/react';
import { renderWithRouter } from '@/utils/renderWithRouter';
import Service from '@/components/screens/home/service/Service';
import * as router from 'react-router';


describe('Service', () => {
  const navigate = jest.fn();

  beforeEach(() => {
    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Service should renders correctly', () => {
    renderWithRouter(<Service />);
    expect(screen.getByTestId('service')).toMatchSnapshot();
  });
  it('Button should navigate user in catalog page', () => {
    renderWithRouter(<Service />);
    const buttons = screen.getAllByText('Купить');
    const button = buttons[0];
    fireEvent.click(button);
    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith('/catalog');
  });
});
