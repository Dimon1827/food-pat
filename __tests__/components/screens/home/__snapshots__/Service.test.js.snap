// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service Service should renders correctly 1`] = `
<section
  class="service"
  data-testid="service"
>
  <h2
    class="visually-hidden"
  >
    Услуги
  </h2>
  <div
    class="container"
  >
    <ul
      class="service__list"
    >
      <li
        class="service__item"
      >
        <p
          class="service__description"
        >
          Лучшие органические продукты онлайн
        </p>
        <button
          class="service__link"
        >
          Купить
        </button>
      </li>
      <li
        class="service__item"
      >
        <p
          class="service__description"
        >
          Приготовьте свой завтрак Здоровый и легкий
        </p>
        <button
          class="service__link"
        >
          Купить
        </button>
      </li>
    </ul>
  </div>
</section>
`;
